src/BugsnagUnityForReview.mm

```diff
@@ -0,0 +1,876 @@
+  NSMutableSet *ns_redactedKeys = [NSMutableSet new];
+  for (int i = 0; i < count; i++) {
+    const char *key = redactedKeys[i];
+    if (key != nil) {
+      NSString *ns_key = [NSString stringWithUTF8String: key];
+      [ns_redactedKeys addObject: ns_key];
+    }
+  }
+  ((__bridge BugsnagConfiguration *)configuration).redactedKeys = ns_redactedKeys;
+}
+
+void bugsnag_setAutoNotifyConfig(const void *configuration, bool autoNotify) {
+  ((__bridge BugsnagConfiguration *)configuration).autoDetectErrors = autoNotify;
+}
+
+void bugsnag_setAutoTrackSessions(const void *configuration, bool autoTrackSessions) {
+  ((__bridge BugsnagConfiguration *)configuration).autoTrackSessions  = autoTrackSessions;
+}
+
+void bugsnag_setPersistUser(const void *configuration, bool persistUser) {
+  ((__bridge BugsnagConfiguration *)configuration).persistUser = persistUser;
+}
+
+void bugsnag_setSendLaunchCrashesSynchronously(const void *configuration, bool sendLaunchCrashesSynchronously) {
+  ((__bridge BugsnagConfiguration *)configuration).sendLaunchCrashesSynchronously = sendLaunchCrashesSynchronously;
+}
+
+void bugsnag_setEndpoints(const void *configuration, char *notifyURL, char *sessionsURL) {
+  if (notifyURL == NULL || sessionsURL == NULL)
+    return;
+
+  NSString *ns_notifyURL = [NSString stringWithUTF8String: notifyURL];
+  NSString *ns_sessionsURL = [NSString stringWithUTF8String: sessionsURL];
+
+  ((__bridge BugsnagConfiguration *)configuration).endpoints = [[BugsnagEndpointConfiguration alloc] initWithNotify:ns_notifyURL sessions:ns_sessionsURL];
+}
+
+void bugsnag_setMetadata(const char *section, const char *jsonString) {
+
+  if (section == NULL || jsonString == NULL)
+  {
+    return;
+  }
+
+  NSString *tabName = [NSString stringWithUTF8String: section];
+
+ [Bugsnag.client addMetadata:getDictionaryFromMetadataJson(jsonString) toSection:tabName];
+
+}
+
+const char * bugsnag_retrieveMetaData() {
+    return getMetadataJson([Bugsnag.client metadata].dictionary);
+}
+
+void bugsnag_removeMetadata(const void *configuration, const char *tab) {
+  if (tab == NULL)
+    return;
+
+  NSString *tabName = [NSString stringWithUTF8String:tab];
+  [Bugsnag.client clearMetadataFromSection:tabName];
+}
+
+void bugsnag_startBugsnagWithConfiguration(const void *configuration, char *notifierVersion) {
+  if (notifierVersion) {
+    ((__bridge BugsnagConfiguration *)configuration).notifier =
+      [[BugsnagNotifier alloc] initWithName:@"Unity Bugsnag Notifier"
+                                    version:@(notifierVersion)
+                                        url:@"https://github.com/bugsnag/bugsnag-unity"
+                               dependencies:@[[[BugsnagNotifier alloc] init]]];
+  }
+  [Bugsnag startWithConfiguration: (__bridge BugsnagConfiguration *)configuration];
+  // Memory introspection is unused in a C/C++ context
+  [BSG_KSCrash sharedInstance].introspectMemory = NO;
```

Comment:
This is not necessary - it's set to NO by default now.

```suggestion
  // Memory introspection is unused in a C/C++ context
  [BSG_KSCrash sharedInstance].introspectMemory = NO;
```

Reply:
Removed 👍